gpu/octree/include/pcl/gpu/octree/octree.hpp

```diff
@@ -135,11 +144,39 @@ namespace pcl
               * \param[in] centers array of centers  
               * \param[in] indices indices for centers array (only for these points search is performed)
               * \param[in] radius radius for all queries
-              * \param[in] max_results max number of returned points for each querey
+              * \param[in] max_results max number of returned points for each query
               * \param[out] result results packed to single array
               */
             void radiusSearch(const Queries& centers, const Indices& indices, float radius, int max_results, NeighborIndices& result) const;
 
+            /** \brief Performs batch radius search on GPU
+              * \param[in] centers array of centers
+              * \param[in] radius radius for all queries
+              * \param[in] max_results max number of returned points for each query
+              * \param[out] result results packed to single array
+              * \param[out] sqr_distances corresponding squared distances to query point
+              */
+            void radiusSearch(const Queries& centers, const float radius, const int max_results, NeighborIndices& result, ResultSqrDists& sqr_distances) const;
+
+            /** \brief Performs batch radius search on GPU
+              * \param[in] centers array of centers
+              * \param[in] radiuses array of radiuses
+              * \param[in] max_results max number of returned points for each query
+              * \param[out] result results packed to single array
+              * \param[out] sqr_distances corresponding squared distances to query point
+              */
+            void radiusSearch(const Queries& centers, const Radiuses& radiuses, const int max_results, NeighborIndices& result, ResultSqrDists& sqr_distances) const;
```

Comment:
the plural of radius is radii. Hows does this work with multiple centers and radii? is it a 1-to-1 relationship i.e. there's the same amount of radii as query points?

Reply:
Yes, its a 1 to 1 relationship, and the same amount of radii is expected. (there is no check for this)

I think modifying this variable name would also require modifying the name of the pre-existing function for consistency as well, which is in public API. Would that be okay? (Technically radii and radiuses are both accepted as the plural form i think :P )

Reply:
> (Technically radii and radiuses are both accepted as the plural form i think :P )

didn't know. Just leave it then.

Reply:
> Yes, its a 1 to 1 relationship, and the same amount of radii is expected. (there is no check for this)

Can you make the docstring reflect that? And also potential add an `assert` in the code, so that the check is there in debug mode.